build-support/bin/pre-commit.sh

```diff
@@ -9,11 +9,40 @@ then
     export GIT_HOOK=1
 fi
 
+DIRS_TO_CHECK=(
+  src
+  tests
+  pants-plugins
+  examples
+  contrib
+)
+
 # TODO(#7068): Fix all these checks to only act on staged files with
-# `git diff --cached --name-only`! See check_header.sh for an example of this command.
-echo "* Checking packages" && ./build-support/bin/check_packages.sh || exit 1
-echo "* Checking headers" && ./build-support/bin/check_header.sh || exit 1
-echo "* Checking for banned imports" && ./build-support/bin/check_banned_imports.sh || exit 1
+# `git diff --cached --name-only`!
+
+# TODO: test all the scripts in this file in test_git.py, remove uses of `|| exit 1`, and add an
+# integration test!
+set -e
+
+# Read lines of output into the array variable ADDED_FILES, without trailing newlines (-t). See
+# https://www.gnu.org/software/bash/manual/html_node/Bash-Builtins.html#index-readarray.
+# NB: `readarray` is available on bash >=4.
+readarray ADDED_FILES -t < <(./build-support/bin/get_added_files.sh)
+
+echo "* Checking packages"
+# TODO: Determine the most *hygienic* way to split an array on the command line in portable bash,
+# and stick to it.
+./build-support/bin/check_packages.sh "${DIRS_TO_CHECK[@]}"
+
+echo "* Checking headers"
+# Read added files from stdin, and ensure check_header_helper.py checks for the current copyright
+# year for the intersection of these files with the ones it checks.
+# Exporting IGNORE_ADDED_FILES will avoid checking the specific copyright year for added files.
```

Comment:
We should consider renaming this to `PANTS_IGNORE_ADDED_FILES` for consistency with our other env variables and to avoid namespace conflicts.

IIRC you introduced the variable in the last PR, but better to fix now than never.

Reply:
This is a REALLY good idea, doing that now.

Reply:
Done!!!